### DISCUSSION TITLE
Roadmap for integration with OCaml 5.0?

### DISCUSSION THREAD

**Discussion by ghennequin** [owlbarn/owl#628] [General]
@ryanrhymes @jzstark Just wondering, do you guys have a roadmap towards making sure `owl` plays well with OCaml 5.0?

**Comment by jzstark** [comment]
Thanks for asking. We are beginning to migrate to OCaml 5.0 while removing certain parts of Owl. Currently the code compiles and runs fine on 5.0, but we will see if any more change is required.

**Comment by ghennequin** [comment]
Thanks! I was wondering if there are global mutable states (e.g. random number 
generators, AD tape, ...) for which we should provide the user an easy way of 
maintaining thread safety... For RNGs, we could provide a convenient interface 
for each domain to have its own RNG, maybe we can get inspiration from JAX's key 
splitting (?). I'd have to think more about the AD part, it seems non-trivial.
